modules/civiremote_event/src/Controller/RegisterFormController.php

```diff
@@ -16,32 +16,114 @@
 namespace Drupal\civiremote_event\Controller;
 
 use Drupal;
+use Drupal\civiremote_event\CiviMRF;
 use Drupal\civiremote_event\Form\RegisterForm\RegisterFormInterface;
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Access\AccessResultAllowed;
+use Drupal\Core\Access\AccessResultNeutral;
 use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Messenger\MessengerInterface;
+use Drupal\Core\Routing\RouteMatch;
 use stdClass;
+use \Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
 
 class RegisterFormController extends ControllerBase {
 
-  public function form(stdClass $event = NULL, $profile = NULL) {
-    // Use default profile if not given.
-    if (!isset($profile) && isset($event)) {
-      $profile = $event->default_profile;
+  /**
+   * @var CiviMRF $cmrf
+   *   The CiviMRF service.
+   */
+  protected $cmrf;
+
+  /**
+   * @param CiviMRF $cmrf
+   *   The CiviMRF service.
+   */
+  public function __construct(CiviMRF $cmrf) {
+    $this->cmrf = $cmrf;
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @param \Symfony\Component\DependencyInjection\ContainerInterface $container
+   *   The Drupal service container.
+   *
+   * @return static
+   */
+  public static function create(ContainerInterface $container) {
+    /**
+     * Inject dependencies.
+     * @var CiviMRF $cmrf
+     */
+    $cmrf = $container->get('civiremote_event.cmrf');
+    return new static(
+      $cmrf
+    );
+  }
+
+  public function form(RouteMatch $route_match, string $context, stdClass $event = NULL, string $profile = NULL, stdClass $event_token = NULL) {
```

Comment:
It could be stated in phpdoc that `$event_token` contains the event data resolved by the event token and not the event token itself. Maybe it would make sense to have a separate method for the request with event token - as of now the method signature reads like both `$event` and `$event_token` are optional, though one of them is required.